src/components/Identity.vue

```diff
@@ -30,47 +30,210 @@
       </div>
       <div class="form-group row">
         <label :for="urlLabel" class="col-sm-2 col-form-label">URL</label>
-        <div class="col-sm-10">
+        <div class="col-sm-9">
           <input
             type="text"
             class="form-control"
             name="url"
             :tabindex="urlTabIndex"
             v-model="identity.url"
-            @blur="update"
-            @keyup.enter="update"
+            @change="updateUrl"
+            @keyup.enter="updateUrl"
             :id="urlLabel"
             :disabled="disabled"
             placeholder="https://example.com/"
           />
         </div>
+        <div class="col-sm-1">
+          <button
+            type="button"
+            class="btn btn-link float-right expand-identity"
+            :tabindex="expandTabIndex"
+            @click="toggleExpanded"
+          >
+            <span class="fas fa-ellipsis-h"></span>
+          </button>
+        </div>
+      </div>
+      <div :class="{ 'd-none': !expandExtras }">
+        <div class="form-group row">
+          <label :for="tagLabel" class="col-sm-2 col-form-label">Tag</label>
+          <div class="col-sm-9">
+            <input
+              type="text"
+              class="form-control"
+              @change="update"
+              @keyup.enter="update"
+              v-model="identity.tag"
+              :tabindex="tagTabIndex"
+              :id="tagLabel"
+            />
+          </div>
+        </div>
+        <div class="form-group row">
+          <label :for="qualityLabel" class="col-sm-2 col-form-label"
+            >Quality</label
+          >
+          <div class="col-sm-9">
+            <select
+              class="custom-select"
+              v-model.number="identity.quality"
+              @change="update"
+              :tabindex="qualityTabIndex"
+              :id="qualityLabel"
+            >
+              <option value="0">Low</option>
+              <option value="1">Mid</option>
+              <option value="2">High</option>
+            </select>
+          </div>
+          <div class="col-sm-1">
+            <button type="button" class="btn btn-link float-right">
+              <span :class="qualityPreview"></span>
+            </button>
+          </div>
+        </div>
+        <div class="form-group row">
+          <label :for="iconLabel" class="col-sm-2 col-form-label">Icon</label>
+          <div class="col-sm-9">
+            <select
+              class="custom-select"
+              v-model="identity.icon"
+              @change="update"
+              :tabindex="iconTabIndex"
+              :id="iconLabel"
+            >
+              <option value="fas fa-link">web</option>
```

Comment:
I feel exactly the same way. I opted to just directly store the CSS classes in the database because (a) they are easy to data-migrate down the road if we have to, (b) that gets around having to replicate a mapping table in Python and JavaScript.